### PR TITLE
feat: run mac compile in workflow

### PR DESCRIPTION
**Pull request by dl239**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: .github/workflows/cicd.yaml [hunk 1]
```diff
@@ -158,6 +158,7 @@ jobs:
       - uses: actions/checkout@v2
 
       - name: prepare release
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
```
Comment (aceforeverd) [inline .github/workflows/cicd.yaml]: this `if` condition can be omitted

File: .github/workflows/cicd.yaml [hunk 2]
```diff
@@ -168,11 +169,12 @@ jobs:
           echo "OPENMLDB_PREFIX=openmldb-$VERSION-darwin" >> $GITHUB_ENV
 
       - name: enable java on release
-        if: github.event_name == 'push'
+        if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v') }}
```
Comment (aceforeverd) [inline .github/workflows/cicd.yaml]: keep the if expr same as the linux job above

File: .github/workflows/cicd.yaml [hunk 3]
```diff
@@ -168,11 +169,12 @@ jobs:
...
         run: |
           echo "SQL_JAVASDK_ENABLE=ON" >> $GITHUB_ENV
 
       - name: configure
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
```
Comment (aceforeverd) [inline .github/workflows/cicd.yaml]: same here, except `install, package, upload` steps can keep the `if` expr check

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**